core/src/main/scala/kafka/utils/ZkUtils.scala

```diff
@@ -795,22 +795,9 @@ class ZkUtils(val zkClient: ZkClient,
     * client updates the path stat.version gets incremented
     */
   def getSequenceId(path: String, acls: java.util.List[ACL] = DefaultAcls): Int = {
-    try {
+      makeSurePersistentPathExists(path)
```

Comment:
@omkreddy yeah, I think this approach is fine for `getBrokerSequenceId`. However, `getSequenceId` seems like a general method and it could be used for other purposes in the future. Personally, I prefer to implement building blocks in the right way if straightforward to do so.